spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubInboundChannelAdapter.java

```diff
@@ -0,0 +1,53 @@
+package org.springframework.cloud.gcp.pubsub.integration.inbound;
+
+import com.google.cloud.pubsub.spi.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.spi.v1.Subscriber;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.SubscriptionName;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.messaging.MessageHeaders;
+
+/**
+ * Created by joaomartins on 5/19/17.
+ */
+public class PubSubInboundChannelAdapter extends MessageProducerSupport {
+
+  private String projectId;
+  private String subscriptionName;
+  private Subscriber subscriber;
+
+  public PubSubInboundChannelAdapter(String projectId, String subscriptionName) {
+    this.projectId = projectId;
+    this.subscriptionName = subscriptionName;
+  }
+
+  @Override
+  protected void onInit() {
+    super.onInit();
+
+    subscriber = Subscriber.defaultBuilder(SubscriptionName.create(
+        projectId, this.subscriptionName), this::receiveMessage).build();
+    subscriber.startAsync();
```

Comment:
Well, this hard connection logic should be moved to the `doStart()`.
It's not good to perform low-level resource interaction during Spring container initialization.

On the other hand with the `Lifecycle` in hands I even can decide do not connection in the beginning of application life.
Or can decide to `stop()` such a component at runtime.
More info in the Spring Reference Manual: http://docs.spring.io/spring/docs/current/spring-framework-reference/html/beans.html#beans-factory-lifecycle